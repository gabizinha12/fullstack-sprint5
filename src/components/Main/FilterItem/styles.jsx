import styled from "styled-components";

export const ListItem = styled.li`
  display: flex;
  align-items: center;
  padding: 12px 14px;
  &:hover {
    cursor: pointer;
    background-color: rgba(38, 30, 30, 0.04);
  }
`;
export const ItemLabel = styled.span`
  text-transform: uppercase;
  font-size: 0.6875rem;
  font-weight: 400;
  color: #261e1e;
`;
export const ItemImage = styled.img`
  height: 20px;
  margin-left: 8px;
`;
