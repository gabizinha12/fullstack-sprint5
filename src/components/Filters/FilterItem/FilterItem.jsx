import FilterIcon from "../../../assets/filter.svg";
import React from "react";
import { ListItem, ItemLabel, ItemImage } from "./styles";

export function FilterItem({ label }) {
  return (
    <ListItem>
      <ItemLabel>{label}</ItemLabel>
      <ItemImage src={FilterIcon} alt="Ã­cone de filtro" />
    </ListItem>
  );
}
