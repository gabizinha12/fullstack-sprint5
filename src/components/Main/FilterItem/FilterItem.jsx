import FilterIcon from "../../assets/filter.svg";
import React from "react";
import { ListItem, ItemLabel, ItemImage } from "./styles";

function FilterItem({ label }) {
  return (
    <ListItem>
      <ItemLabel label={label} />
      <ItemImage src={FilterIcon} alt="Ã­cone de filtro" />
    </ListItem>
  );
}

export default FilterItem;
