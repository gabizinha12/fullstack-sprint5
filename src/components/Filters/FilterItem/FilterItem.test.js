import { render, screen } from "@testing-library/react";
import { FilterItem } from "./FilterItem";

describe("Teste de snapshot FilterItem", () => {
  it("should render FilterItem", () => {
    const { container } = render(<FilterItem label="abc" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
