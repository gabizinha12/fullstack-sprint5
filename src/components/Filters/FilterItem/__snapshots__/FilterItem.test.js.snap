// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de snapshot FilterItem should render FilterItem 1`] = `
<li
  class="sc-bdnxRM kYenBY"
>
  <span
    class="sc-gtsrHT bAmSmd"
  >
    abc
  </span>
  <img
    alt="Ã­cone de filtro"
    class="sc-dlnjwi ckhPXh"
    src="filter.svg"
  />
</li>
`;
